test/topology/test_coordinator_queue_management.py

```diff
@@ -5,12 +5,20 @@
 #
 from test.pylib.manager_client import ManagerClient
 from test.topology.conftest import skip_mode
+from collections.abc import Coroutine
 import pytest
 import logging
 import asyncio
 
 logger = logging.getLogger(__name__)
 
+async def wait_for_first_completed(coros: list[Coroutine]):
+    done, pending = await asyncio.wait([asyncio.create_task(c) for c in coros], return_when = asyncio.FIRST_COMPLETED)
+    for t in pending:
+        t.cancel()
+    for t in done:
+        await t
```

Comment:
Nice, I also noticed the warning about unwaited coroutines but decided to not wait for it.

Reply:
This part isn't about the warnings. It's an error check. Without doing the `await`, you only know that the future has resolved, but you don't know if it was successful or not.

If you are waiting for a log entry to show up, but the wait times out, you want the test to fail, not to move on.

Reply:
I think that even if you didn't do an await here, there should be no warnings about unused coroutines, because the coroutines *are* spawned into tasks, and I think that's enough to consider them used.

But the old code didn't even turn them into tasks.

Reply:
I'm talking about the cancel(). Yes, I placed my comment inaccurately.